scene/main/viewport.cpp

```diff
@@ -1868,10 +1868,9 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 
 				// If the mouse is over a menu button, this menu will open automatically
 				// if there is already a pop-up menu open at the same hierarchical level.
-				if (popup_menu_parent && menu_button &&
-						popup_menu_parent->get_icon().is_null() &&
-						menu_button->get_icon().is_null() &&
-						(popup_menu->get_parent()->get_parent()->is_a_parent_of(menu_button) ||
+				if (popup_menu_parent && menu_button && popup_menu_parent->is_switch_on_hover() &&
+						!menu_button->is_disabled() && menu_button->is_switch_on_hover() &&
+						(popup_menu_parent->get_parent()->is_a_parent_of(menu_button) ||
```

Comment:
Don't change that or popup menus of script editor will not work.

Reply:
keep this line with two `get_parent()` like that:
`(popup_menu->get_parent()->get_parent()->is_a_parent_of(menu_button) ||`

Reply:
The part that affects the Script Editor menus is `menu_button->get_parent()->is_a_parent_of(popup_menu)`.

`popup_menu->get_parent()->get_parent()` and `popup_menu_parent->get_parent()` are exactly the same thing.

Reply:
This is not the same thing, See here:
![peek 04-08-2018 08-37](https://user-images.githubusercontent.com/1387165/43676151-af511f56-97c1-11e8-8d10-5113234232a3.gif)
I can't select "Search" and "Edit" with your changes. Pls, fix this.

Reply:
Sry, you're already using parent. Then changes are not needed.